ir/value.cpp

```diff
@@ -199,10 +201,15 @@ StateValue Input::toSMT(State &s) const {
     val = expr::mkIf(type.extract(0, 0) == 0, val, undef);
   }
 
-  if (has_deref) {
+  if (has_deref || has_nonnull) {
     Pointer p(s.getMemory(), val);
-    s.addAxiom(p.isDereferenceable(attrs.getDerefBytes(), bits_byte/8, false));
-    s.addAxiom(type == 0);
+    if (has_deref) {
+      s.addAxiom(type == 0);
+      s.addAxiom(p.isDereferenceable(attrs.getDerefBytes(), bits_byte/8, false));
+    }
+    if (has_nonnull && !has_deref) {
+      s.addAxiom(type.extract(1, 1) == 0);
```

Comment:
why allow nonnull arguments to be undef?
A function call with undef as argument is UB, so I think we can assume nonnull arguments aren't either poison or undef.

Reply:
Because an input can be partially undef with non-null value I think.
Memory::mkUndefInput has a nice way of encoding non-null undef value.